src/workerd/api/system-streams.h

```diff
@@ -31,6 +32,18 @@ kj::Own<WritableStreamSink> newSystemStream(
 //
 // NOTE: As with the other overload of newSystemStream(), `inner` must be wholly owned.
 
+struct SystemMultiStream {
+  kj::Own<ReadableStreamSource> readable;
+  kj::Own<WritableStreamSink> writable;
+};
+
+SystemMultiStream newSystemMultiStream(
+    kj::Own<PipelinedAsyncIoStream> rc, StreamEncoding encoding,
```

Comment:
Hmm do we actually expert to support gzip-encoding layered directly on TCP sockets? I would sort of expect instead we'd leave it up to userspace to add a `CompressionStream` if they want it.

Or did you just add the `encoding` parameter here because it was effectively free to implement, since you're reusing the existing code?

Reply:
> Or did you just add the encoding parameter here because it was effectively free to implement, since you're reusing the existing code?

Yep, this. I'll likely end up refactoring this to not use the Encoded streams after speaking with Harris (that will be done in https://github.com/cloudflare/workerd/pull/205)